### ISSUE TITLE
TACL doesn't save many sites if 3rd party trackers are blocked

### ISSUE THREAD

**Issue by bendover22** [ettoolong/TextareaCache#29]
In Fx 55.0.2, suddenly TACL 2.3 stopped saving any new entries.  Had worked fine for several Fx 54 & 55 versions.  Thought, maybe the file's too big (but didn't seem so).  Manually deleted older entries from textareacache.json & saved.  

I can create a clean profile & try it, but in this profile (few wks old), TACL worked great, then not at all,  back to working "some" and finally not at all.  
UPDATE:  I may have found the problem.  When I disabled uBlock Origin just now on Github, TACL started saving - including saving what I wrote before turning uBo off,
Might depend on whether uBo's filters **happen** to allow the correct elements or 3rd parties on a site, or not - for TACL to work correctly.

uBo wasn't allowing githubapp.com & was blocking one element from  github.com

I can't upload screen shots on Github (this page, anyhow) - by  d&d, c&P, or Select a file - even w/ uBo turned off.
Does uploading images / files require allowing "github-production-user...amazonaws.com?

Back to TACL working on some sites - not others.  Maybe 1 month ago while using TAC (not Lite), I don't ever remember it not saving typing on any site, regardless of whether 3rd party trackers / scripts were allow or not.  Generally,most  3rd parties were NOT allowed, while still used TAC & it worked.  But now it's a different animal.  I understand.

But I can't allow every 3rd party tracker (not always nice guys), just so TACL works.
Maybe there's a compromise?  I feel sorry for Mozilla addon developers right now.  I also feel sorry for users.

**Comment by wildskyf** [comment]
Hi @bendover22,

Has it still happened in the latest version of Textarea Cache (v2.5.0)?

**Comment by bendover22** [comment]
> Has it still happened in the latest version of Textarea Cache (v2.5.0)?
Yes and no.  It seemed to start saving a couple of sites (I regularly visit) that it didn't before.
I assume could be due to changes in v2.5.0, or not.

Other sites still have problems.  Not sure if some sites now having problems in v2.5 also did in the previous version.
On this Github page, as of the last sentence, "...previous version," it wasn't saving anything.
There's not (wasn't) an entry in the drop list, that lists all saved posts with each date.

UNTIL... I disabled "blocking of remote fonts" in uBlock Origin 1.14.16 (web ext version).
Though I have uBo completely **disabled** for Github- NOT blocking any trackers, apparently some  of  the 4 icons / features at the bottom of uBo's main popup UI can be active.  When I removed the "blocking of remote fonts" on this page, immediately what I had already typed appeared in TACL. 

 When I tried to duplicate the problem, I couldn't.  Closed all Github tabs, cleared all cookies & cache, logged back into Github.  TACL kept recording all I typed & kept up with erasures & re-writes.  I re-enabled all 4 of uBo's  per site blocking options - still couldn't get TACL to stop recording.

But earlier this morning, several posts on Sourceforge, TACL only recorded maybe one short sentence out of several paragraphs.  That's on several different posts.
So I don't know what's going on, except the same uBo settings existed on Sourceforge as for Github.  I'll go back to sourceforge later - see if I can duplicate.

I'm thinking this is a bug (don't know which addon) that only happens under specific cases & certain sites.  Once you change some settings, maybe it works until actually restart Firefox?

 See: https://github.com/gorhill/uBlock/wiki/Per-site-switches#no-remote-fonts  
UBo is disabled (for trackers / 1st and 3rd party scripts), but the 4 "per site switches" can still be active (or turned off).

**Comment by bendover22** [comment]
On the uBlock Origin forum at https://discourse.mozilla.org/t/support-ublock-origin
TACL didn't save any text.  All trackers or blocking was turned off in NoScript & uBo.  Otherwise, the site wouldn't show a "Login with Email" button.  That's a strange site.  
Just trying to give some sites that don't work - if you can spot similarities.

**Comment by wildskyf** [comment]
So now, we know if a website is remote-font-blocked when user visite site first time, textarea cache saving will fail.
If you enable remote font, saving will work even if you disable it again.

**Comment by bendover22** [comment]
On Github, I couldn't post the one above this, unless "Cosmetic Filtering" (the icon at menu bottom, looks like a sideways (American) football.  If cosmetic filtering was on, got msg, "sorry, you can't post right now," when I submitted - no explanation why.  

When I disabled cosmetic filtering, it took it.  I'm assuming that was the reason & not that the site / server was too busy at the moment?
The only trackers or JS turned off in NoScript was google analytics.
The only js blocked in uBo was "githubapp.com" & google analytics.
